src/codegen/generators/variable_generator.rs

```diff
@@ -115,8 +115,16 @@ impl<'ctx, 'b> VariableGenerator<'ctx, 'b> {
         let type_name = global_variable.get_type_name();
         let variable_type = self.types_index.get_associated_type(type_name)?;
 
-        let mut global_ir_variable =
-            self.llvm.create_global_variable(self.module, global_variable.get_name(), variable_type);
+        let name = if self
+            .global_index
+            .get_type_information_or_void(type_name)
+            .is_enum() {
+                global_variable.get_qualified_name()
+            } else {
+                global_variable.get_name()
+            };
+
+        let mut global_ir_variable = self.llvm.create_global_variable(self.module, name, variable_type);
         if linkage == LinkageType::External {
             global_ir_variable = global_ir_variable.make_external();
```

Comment:
can enums be linked externally? if so, I'll need to exclude externally linked enums in the above conditional

Reply:
you can have enums as part of a library so I would say yes

Reply:
Just out of curiousity, how would the declaration look like?

Compiling the following on master (taking a guess on this declaration):
```
{external}
VAR_GLOBAL
    myEnum: (a,b);
END_VAR
```
results in this IR:
```
@myEnum= external global i32
@a = unnamed_addr constant i32 0
@b = unnamed_addr constant i32 1
```
The variants are defined as local constants - wouldn't they clash with the external ones, if it had different values?
What would happen if the linked enum looked like this:
```
myEnum: (a := 29, b);
```

Reply:
I went ahead and tried it:
enum "library":

extEnum.st
```
TYPE extEnum : (a := 29, b);
END_TYPE
```

compiled with `cargo r -- target/extEnum.st --shared -o libextEnum.so`

demo.st
```
{external}
VAR_GLOBAL
    Variable: extEnum; (* Comment *)
END_VAR

FUNCTION main : DINT
    VAR
        x: DINT;
    END_VAR

    x := extEnum.a;
END_FUNCTION
```

compiled and linked with `LD_LIBRARY_PATH=. cargo r -- target/demo.st -i target/extEnum.st -lextEnum -L. --ir`

resulting IR:
master
```
; ModuleID = '/tmp/.tmpCBozkp/target/demo.st.ll'
source_filename = "target/demo.st"

@Variable = external global i32
@a = constant i32 29
@b = unnamed_addr constant i32 30

define i32 @main() {
entry:
  %main = alloca i32, align 4
  %x = alloca i32, align 4
  store i32 0, i32* %x, align 4
  store i32 0, i32* %main, align 4
  store i32 29, i32* %x, align 4
  %main_ret = load i32, i32* %main, align 4
  ret i32 %main_ret
}
```

this branch
```
; ModuleID = '/tmp/.tmpVy0Kpa/target/demo.st.ll'
source_filename = "target/demo.st"

@Variable = external global i32
@extEnum.a = constant i32 29
@extEnum.b = unnamed_addr constant i32 30

define i32 @main() {
entry:
  %main = alloca i32, align 4
  %x = alloca i32, align 4
  store i32 0, i32* %x, align 4
  store i32 0, i32* %main, align 4
  store i32 29, i32* %x, align 4
  %main_ret = load i32, i32* %main, align 4
  ret i32 %main_ret
}

```

It looks like the constants are resolved correctly with both approaches, so not differentiating should be fine?

Reply:
Ok, lt looks like since this is a constant the folding is happening correctly and therefore no conflicts arise. My assumption was this will behave similarly to a global variable being defined twice which is where the external global ... comes into play

Reply:
Maybe I misunderstood something but wouldn't this commit now ensure the enums will not conflict? Even if they are external? Why would the linking fail if the enums have their own unique names?

Reply:
Let me try to rephrase:
Externally defined enum types work as expected, my concern were inline enums:
Inline declared enums would take the value of the locally defined variant, even when decorated with the external pragma.
I thought it would be better to validate against this instead of potentially have misleading behaviour.
However, I noticed that if the externally declared enum is also an inline enum, the changes from #1175 now detect it as ambigious datatype during validation, so this point is moot.